### ISSUE TITLE
Wow this is cool!

### ISSUE THREAD

**Issue by JordanMarr** [edgarfgp/Fabulous.AST#60]
Can’t believe I didn’t know about this.

**Comment by edgarfgp** [comment]
Thanks @JordanMarr. I’m close to 1.0 and now need to write some docs, let me know if you have a use case for this lib

**Comment by JordanMarr** [comment]
I was going to see if I can rewrite the SqlHydra.Cli code gen with it. I am already a few versions out of date with the AST and have been putting off upgrading because it’s such a puzzle. 

I’m doing modules with records and some misc functions as well. 

https://github.com/JordanMarr/SqlHydra/blob/v2_4_nullable/src/SqlHydra.Cli/SchemaGenerator.fs

**Comment by edgarfgp** [comment]
> I was going to see if I can rewrite the SqlHydra.Cli code gen with it. I am already a few versions out of date with the AST and have been putting off upgrading because it’s such a puzzle.
> 
> I’m doing modules with records and some misc functions as well.
> 
> https://github.com/JordanMarr/SqlHydra/blob/v2_4_nullable/src/SqlHydra.Cli/SchemaGenerator.fs
> 
> This is an example of what gets generated: https://github.com/JordanMarr/SqlHydra/blob/v2_4_nullable/src/Tests/SqlServer/AdventureWorksNet8.fs
> 
> Although I do have a few places where I cheat and splice pre-baked strings in as well.

Yeah you should be able to remove a lot of that boilerplate using this DSL

Will be releasing a new version on Friday so you can give it a try 🙂

**Comment by JordanMarr** [comment]
Looking forward to it! I messed around with it last night, but hit a few snags with trying to nest a module inside the namespace. 
I thought I would wait for the latest update before getting too deep into it.

**Comment by edgarfgp** [comment]
> Looking forward to it! I messed around with it last night, but hit a few snags with trying to nest a module inside the namespace. I thought I would wait for the latest update before getting too deep into it.

This would possible in the next version in the form of 

```fsharp
    [<Test>]
    let ``Produces a namespace namespace with nested modules`` () =
        Namespace("X") {
            NestedModule("A") {
                Value("x", "12", false)
            }
            NestedModule("B") {
                Value("x", "12", false)
            }
        }
        |> produces
            """
namespace X

module A =
    let x = 12

module B =
    let x = 12
"""
```

**Comment by edgarfgp** [comment]
@JordanMarr Releases `0.8.0` https://www.nuget.org/packages/Fabulous.AST. Please look at the unit test in this project while I add some docs samples

**Comment by JordanMarr** [comment]
I was able to generate a namespace with a nested module and some records. 🎉
Very exciting!

**Comment by JordanMarr** [comment]
One thing I'm struggling with is how to make record fields handle field names with spaces.

Ex:

```F#
Record("Person") {
    Field("First Name", "string")
}
```

Which I would expect would produce this:
```F#
type Person  {
    ``First Name``: string
}
```

But currently produces this:
```F#
type Person  {
    First Name: string
}
```

**Comment by edgarfgp** [comment]
@JordanMarr This should be fixed in https://www.nuget.org/packages/Fabulous.AST/0.8.1
